incubator/hnc/api/v1alpha1/hierarchical_namespace.go

```diff
@@ -0,0 +1,67 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HNSStatus describes the status of a hierarchical namespace. The status could be
+// "missing", "ok" or "conflict". The definitions will be described below.
+type HNSStatus string
```

Comment:
Right now, this would give us `status.status`, which is a bit odd. Pods and containers use `state`, eg:

```
status:
  state: ok | missing | conflict
```

I'd recommend following that, sgty?